diff --git a/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm b/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm
index 2a677974ca8a929849109f294bddcb239bb202d0..f695556b1f3a5559d63c26288e17fbc1a7bbc05a 100644
--- a/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm
+++ b/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm
@@ -435,7 +435,15 @@ void ObjCTurboModule::performVoidMethodInvocation(
     @try {
       [inv invokeWithTarget:strongModule];
     } @catch (NSException *exception) {
-      throw convertNSExceptionToJSError(runtime, exception, std::string{moduleName}, methodNameStr);
+      if (shouldVoidMethodsExecuteSync_) {
+        throw convertNSExceptionToJSError(runtime, exception, std::string{moduleName}, methodNameStr);
+      }
+      RCTLogError(
+          @"TurboModule '%s' void method '%s' threw: %@ - %@",
+          moduleName,
+          methodName,
+          exception.name,
+          exception.reason);
     } @finally {
       [retainedObjectsForInvocation removeAllObjects];
     }
